### PR TITLE
Separate Components into Presentational and Containers - fixes #234

### LINKED ISSUE #234: Separate components into Presentational and Container
As recommended in the [Redux documentation](http://redux.js.org/docs/basics/UsageWithReact.html), it would be nice to separate our components into Presentational (in charge only of how things look) and Container (in charge of how things work) components.

That'd make for a more robust codebase and less dependent components. Please read this [article from Dan Abramov](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0#.clyv6nbnv) for more context.

That'd mean `connect()`, `mapStateToProps()`, `mapDispatchToProps()`, will be moved to another (smarter) component, which will then feed our presentational components with just the data they need.

### PR DESCRIPTION
**Pull request by kdemoya**
Separated components so presentational components don't depend on state or context, their sole purpose now is to render and behave based on passed props. Container will handle the rest.

View #234 for more info.

### CODE REVIEW

**Comment by bernardodiasc** [comment]
Nice!

File: app/containers/GoalsContainer.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/**
+ * Unleash | GoalsContainer.js
```
Comment (BartKrol) [inline app/containers/GoalsContainer.js]: Btw, do we actually need those comments? Imo they are redundant :P

Comment (karolsojko) [inline app/containers/GoalsContainer.js]: I don't think we need them - probably auto-generated @kdemoya ?

Comment (kdemoya) [inline app/containers/GoalsContainer.js]: Yeah, I got my IDE to generate those, can remove them if needed. 😛

**Pull request merged**